packages/react/src/components/numeric-input/use-numeric-input.tsx

```diff
@@ -0,0 +1,164 @@
+    defaultValue,
+    max,
+    min,
+    onBlur,
+    onChange,
+    precision,
+    required,
+    invalid,
+    validationErrorMessage,
+    value,
+}: UseNumericInputParams): UseNumericInputReturn {
+    const { t } = useTranslation('numeric-input');
+    const [stateValue, setStateValue] = useState(
+        validateProvidedValue(value?.toString() ?? defaultValue?.toString() ?? ''),
+    );
+    const [stateErrorMessage, setStateErrorMessage] = useState<string | undefined>();
+
+    const isInvalid = invalid ?? (validationErrorMessage !== undefined || stateErrorMessage !== undefined);
+    const errorMessage = (validationErrorMessage ?? stateErrorMessage);
+
+    const validate = useCallback((inputValue: string): void => {
+        let error: string | undefined;
+
+        if (required && inputValue === '') {
+            error = t('requiredValidationErrorMessage');
+        } else if (inputValue !== '') {
+            if (max !== undefined && Number(inputValue) > max) {
+                error = t('maxValidationErrorMessage', { max });
+            } else if (min !== undefined && Number(inputValue) < min) {
+                error = t('minValidationErrorMessage', { min });
+            }
+        }
+
+        setStateErrorMessage(error);
+    }, [max, min, required, t]);
+
+    const createCallbackData = (inputValue: string): NumericInputCallbackData => ({
+        value: inputValue,
+        valueAsNumber: inputValue === '' ? null : Number(inputValue),
+    });
+
+    const handlePaste = useCallback((event: ClipboardEvent<HTMLInputElement>): void => {
+        event.preventDefault();
+        let newValue = event.clipboardData.getData('text/plain');
+
+        newValue = cleanPastedContent(newValue);
+
+        if (precision !== undefined) {
+            newValue = truncateAtPrecision(newValue, precision);
+        }
+
+        // When the content is invalid, we will clear the field so the user has a feedback that the paste didn't work
+        if (!isValidValueForInput(newValue)) {
+            newValue = '';
+        }
+
+        setStateValue(newValue);
+    }, [precision]);
+
+    const handleChange = useCallback((event: ChangeEvent<HTMLInputElement>): void => {
+        const inputValue = event.target.value;
+        if (inputValue !== '') {
+            if (!isValidValueForInput(inputValue)) {
+                return;
+            }
+            if (precision !== undefined && !isWithinPrecision(inputValue, precision)) {
+                return;
+            }
+        }
+
+        setStateValue(inputValue);
+        onChange?.(event, createCallbackData(inputValue));
+    }, [onChange, precision]);
+
+    const handleBlur = useCallback((event: FocusEvent<HTMLInputElement>): void => {
+        const inputValue = cleanIncompleteNumber(event.target.value);
+        setStateValue(inputValue);
+        validate(inputValue);
+
+        onBlur?.(event, createCallbackData(inputValue));
+    }, [onBlur, validate]);
+
+    useEffect(() => {
```

Comment:
Est-ce que le `useEffect` est nécessaire? Je crois que ça serait un bon use-case pour utiliser un state qui stocke la valeur précédente:
```typescript
const [previousValue, setPreviousValue] = useState(value);

if (value !== undefined && value !== previousValue) {
    setStateValue(validateProvidedValue(value));
    setPreviousValue(value);
}
```

https://react.dev/learn/you-might-not-need-an-effect#adjusting-some-state-when-a-prop-changes
https://react.dev/reference/react/useState#storing-information-from-previous-renders

Reply:
Va vraiment falloir que j'intègre cette réflexion là à mes reviews!

Reply:
Ca va effectivement mieux.